docs/decisions/infra/0005-database-migration-architecture.md

```diff
@@ -0,0 +1,129 @@
+* Cost
+
+## Considered Options
+
+* Execute migrations via direct database connection from Github Actions
+* Execute migrations via a Lambda Function
+* Execute migrations via an ECS Fargate Task
+
+## Decision Outcome
+
+Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences <!-- optional -->
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences <!-- optional -->
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Execute via a Direct Database Connection in Github Action
+
+In this method, the github action runner connects to the database directly to run migrations as part of the ci/cd workflow. There is no cost associated with connecting to the database via github actions.
+
+This method gets the necessary permissions, packages, and versions from the github action runner and the necessary scripts can be written in the Makefile.  <!-- optional -->
+
+#### Pros
+This approach uses existing scripts from the application codebase and is quick and simple to set up.
+
+#### Cons
+
+
+### Execute via a Lambda Function
+
+In this method, a Lambda function is used as the primary method of migration. Either the Lambda is the only compute infrastructure, or it works in conjunction with an EC2 instance or ECS task running something like Flyway. The Lambda needs some sort of stateful supporting infrastructure (like S3 or EC2) for saving the state of the migration, as Lambdas are inherently stateless.
+
+#### Infrastructure Required
+
+- Lambda function to run the migration code
+- Log group for storing logs
+- IAM role for the Lambda to use
+- Some other infrastructure to store the state of the migration (S3, EC2, ECS)
+- Any associated IAM roles for the state-storing infrastructure to use
+- Security group that allows connection to the database
+- IAM role that Github can assume to run the task
+- (optional) A queueing method for the Lambda to feed off of, like SQS
+
+#### Pros
+
+Lambda functions can be very cost effective and fast, as well as simple to use.
+
+Lambdas are low maintenance, and don't stick around after the task.
+
+Even though this solution requires additional support, the compute portion of the task run by AWS Lambda is very simple.
+
+#### Cons
+
+By nature, database migrations need to keep track of state--changes between the previous version, the next version, and at what point the migration is currently at. Lambda functions are stateless by design, so an additional component will be needed to keep track of state.
+
+Lambdas have a comparatively short running time. Either find a way to split up the migrations into small tasks, or extend the running time of the Lambda. This might not scale well with large changesets.
```

Comment:
> Lambdas have a comparatively short running time.

a short runtime isn't a problem. i think what you meant to say is that "lambdas have [max runtime limit of 15 minutes](https://aws.amazon.com/about-aws/whats-new/2018/10/aws-lambda-supports-functions-that-can-run-up-to-15-minutes/) which means that a lambda approach disallows very long running migrations"

Reply:
Yep! I meant that's a shorter running time than an ECS task or an EC2 instance. I'll update this to make it more clear